rspace/src/main/scala/coop/rchain/rspace/merger/StateChange.scala

```diff
@@ -128,8 +128,22 @@ object StateChange {
             .compile
             .drain
       produceChanges <- datumsDiffRef.get
+      _ <- new Exception("State change compute logic error: empty channel change for produce.")
```

Comment:
The logic behind this exception is that we have to (and are) able to determine that channel is adjusted from event log. And now I see that this works as expected: only necessary events are used to compute state change, an  changes on channels involved are always non empty. Which makes sure that `consumesAffected ` and `producesAffected`  from MergingLogic are correct, which gives more confidence that overall logic does not have flaws.